iam-login-service/src/main/java/it/infn/mw/iam/api/requests/GroupRequestUtils.java

```diff
@@ -0,0 +1,145 @@
+package it.infn.mw.iam.api.requests;
+
+import static it.infn.mw.iam.core.IamGroupRequestStatus.APPROVED;
+import static it.infn.mw.iam.core.IamGroupRequestStatus.PENDING;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Strings;
+
+import it.infn.mw.iam.api.account.AccountUtils;
+import it.infn.mw.iam.api.requests.exception.GroupRequestValidationException;
+import it.infn.mw.iam.api.requests.exception.UserMismatchException;
+import it.infn.mw.iam.api.requests.model.GroupRequestDto;
+import it.infn.mw.iam.core.IamGroupRequestStatus;
+import it.infn.mw.iam.persistence.model.IamAccount;
+import it.infn.mw.iam.persistence.model.IamAuthority;
+import it.infn.mw.iam.persistence.model.IamGroup;
+import it.infn.mw.iam.persistence.model.IamGroupRequest;
+import it.infn.mw.iam.persistence.repository.IamAccountRepository;
+import it.infn.mw.iam.persistence.repository.IamGroupRepository;
+import it.infn.mw.iam.persistence.repository.IamGroupRequestRepository;
+
+@Component
+public class GroupRequestUtils {
+
+  @Autowired
+  private IamGroupRequestRepository groupRequestRepository;
+
+  @Autowired
+  private IamAccountRepository accoutRepository;
+
+  @Autowired
+  private IamGroupRepository groupRepository;
+
+  @Autowired
+  private AccountUtils accountUtils;
+
+  private static final IamAuthority ROLE_ADMIN = new IamAuthority("ROLE_ADMIN");
+
+  public Optional<IamGroupRequest> getGroupRequestUuid(String uuid) {
+    return groupRequestRepository.findByUuid(uuid);
+  }
+
+  public IamGroupRequest validateGroupRequestUuid(String uuid) {
```

Comment:
Move validation to java bean validation annotations, and implement validation at the controller level